plugins/catalog-backend/src/database/search.ts

```diff
@@ -28,12 +28,19 @@ const SPECIAL_KEYS = [
   'metadata.generation',
 ];
 
+// The maximum length allowed for search values. These columns are indexed, and
+// database engines do not like to index on massive values. For example,
+// postgres will balk after 8191 byte line sizes.
+const MAX_VALUE_LENGTH = 200;
+
+// Takes any value from an object and turns it into a value string for the
+// search table
 function toValue(current: any): string | null {
   if (current === undefined || current === null) {
     return null;
   }
 
-  return String(current).toLowerCase();
+  return String(current).toLowerCase().substr(0, MAX_VALUE_LENGTH);
```

Comment:
Thoughts on just throwing away too large values instead of truncating?

Reply:
Yeah considered it. Thought maybe the data (if using wildcards) could be useful, buuut... Whatever we do I guess it's gonna be potentially surprising either way

Reply:
Tbh I'm thinking we should not index any fields that could contain any large values, and then just make it a hard error causing the processing to fail

Reply:
I kinda agree. Won't introduce any knowledge of what fields to include / exclude though, I'll just make it skip above a certain length

Reply:
@Rugvip OK this turned into a bit of a refactor where the raw extraction is in one place, row forming (with lowercase and length limit logic) in one place, and the entity structure knowledge in a third place.